aspnetcore/grpc/authn-and-authz.md

```diff
@@ -171,6 +171,43 @@ The preceeding code:
 * Registers the `GreeterClient` type with client factory.
 * Configures the `AuthInterceptor` for this client using `InterceptorScope.Client`. A new interceptor is created for each client instance. When a client is created for a gRPC service or Web API controller, the scoped `ITokenProvider` is injected into the interceptor.
 
+##### Bearer token with gRPC client factory using AsyncAuthInterceptor
+
+If the bearer token can only be generated using asynchronous method calls, the `AsyncAuthInterceptor` must be used.
+
+Assuming the token provider is defined as follows and has been registered as scoped service:
+
+```csharp
+public interface ITokenProvider
+{
+    Task<string> GetToken();
+}
+``` 
+
+You have to register the gRPC client using the `AsyncAuthInterceptor`
+
+```csharp
+services
+    .AddGrpcClient<Greeter.GreeterClient>(o =>
+    {
+        o.Address = new Uri("https://localhost:5001");
+    })
+    .ConfigureChannel((sp, options) =>
+    {
+        var scopeFactory = sp.GetRequiredService<IServiceScopeFactory>();
+        var asyncAuthInterceptor = new AsyncAuthInterceptor(async (context, metadata) =>
+        {
+            using var scope = scopeFactory.CreateScope();
```

Comment:
Hmm, is creating a new scope here, and registering `ITokenProvider` with a scoped lifetime providing any benefit?

Each call is creating a new scope, and each time a new `ITokenProvider` instance is created. Is that much different from a transient lifetime? I haven't built a full gRPC app using scoped auth, but I would have thought the benefit that you're aiming for is caching the token per-request, so calls for that request share it. Will you get that here?

I wonder if there should be a new extension method for the gRPC client factory that is just for configuring `AsyncAuthInterceptor`. An overload can include an `IServiceProvider` that uses the current request scope.

Reply:
I figured this would be necessary if `ITokenProvider` is registered as 'scoped' - or itself has dependencies registered as `scoped` services. If a background process opens an RPC connection (service-2-service communication) we don't necessarily have a scope - then there would be an exception here, right?

Reply:
The example would be shorter and probably easier to understand if we say

> [...] and has been registered as *singleton* service [...]

Reply:
@JamesNK, it is not clear to me what state this is in at the moment.  Does the example require a changed from what it is at this point?